src/Rfc6455/MessageProcessor.php

```diff
@@ -83,6 +82,15 @@ public function onData(string $data, ConnectionInterface $socket, Message $messa
                     try {
                         $frame = new Frame($message->getBuffer());
 
+                        if ($frame->getOpcode() === Frame::OP_CONTINUE && $message->countFrames() === 0) {
+                            throw new ProtocolErrorException('The first frame cannot be a continuation frame');
+                        }
+
+                        if ($frame->getOpcode() === Frame::OP_TEXT && $message->countFrames() >= 1) {
```

Comment:
`!== Frame_CONTINUE` looks better because you may have `OP_BINARY` or other things in the future.